tests/discovery/test_openalex.py

```diff
@@ -0,0 +1,113 @@
+import pytest
+from pytest_regressions.data_regression import DataRegressionFixture
+
+from paperoni.discovery.base import PaperInfo
+from paperoni.discovery.openalex import OpenAlex, QueryError
+
+from ..utils import check_papers, filter_test_papers, iter_affiliations, split_on
+
+PAPERS = [
+    "A Hierarchical Latent Variable Encoder-Decoder Model for Generating Dialogues",
+    "A closer look at memorization in deep networks",
+    "Binarized Neural Networks",
+    "Building End-To-End Dialogue Systems Using Generative Hierarchical Neural Network Models",
+    "Contractive Auto-Encoders: Explicit Invariance During Feature Extraction",
+    "Attention Is All You Need In Speech Separation",
+    "ECAPA-TDNN Embeddings for Speaker Diarization",
+    "GraphMix: Improved Training of GNNs for Semi-Supervised Learning",
+    "HiFormer: Hierarchical Multi-scale Representations Using Transformers for Medical Image Segmentation",
+    "Large-Scale Contrastive Language-Audio Pretraining with Feature Fusion and Keyword-to-Caption Augmentation",
+]
+
+
+@pytest.mark.parametrize(
+    "query_params",
+    [
+        {"institution": "mila"},
+        {"author": "Yoshua Bengio"},
+        {"author_id": "a5086198262"},
+        {"title": "Hierarchical Latent Variable"},
+    ],
+)
+def test_query(data_regression: DataRegressionFixture, query_params: dict[str, str]):
+    discoverer = OpenAlex()
+
+    papers: list[PaperInfo] = sorted(
+        discoverer.query(
+            **query_params,
+            page=1,
+            per_page=100,
+            limit=10000,
+        ),
+        key=lambda x: x.paper.title,
+    )
+
+    papers = list(filter_test_papers(papers, PAPERS))
+
+    assert papers, f"No papers found for {query_params=}"
+
+    match next(iter(query_params.keys())):
```

Comment:
Wrapping this with a for loop on query_params.keys() would make the test robust to combining multiple query parameters.